### ISSUE TITLE
Memory leak with 'use v5.42'

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#10738]
Migrated from [rt.perl.org#78438](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=78438) (status was 'resolved')

Searchable as RT78438$

**Comment by p5pRT** [comment]
### From @nwc10

#### Created by @nwc10



'use v5\.8' no longer leaks memory\. However\, 'use v\.5\.42' still does&#8203;:

$ time bash \-c 'ulimit \-v 70000; \./perl \-Ilib \-e "eval q\{use v5\.42\} while 1;"'
Out of memory\!

real    0m2\.107s
user    0m2\.000s
sys     0m0\.100s

Nicholas Clark

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=low

Site configuration information for perl 5.13.5:

Configured by nick at Tue Oct 19 09:06:10 BST 2010.

Summary of my perl5 (revision 5 version 13 subversion 5) configuration:
  Derived from: d086148c4193252081799d1c5bc16c6cc3f5ae1a
  Platform:
    osname=linux, osvers=2.6.35.4, archname=x86_64-linux-ld
    uname='linux eris 2.6.35.4 #4 smp tue sep 21 09:54:22 bst 2010 x86_64 gnulinux '
    config_args='-Dusedevel=y -Dcc=ccache gcc -Dld=gcc -Ubincompat5005 -Uinstallusrbinperl -Dcf_email=nick@ccl4.org -Dperladmin=nick@ccl4.org -Dinc_version_list=  -Dinc_version_list_init=0 -Doptimize=-Os -Uusethreads -Duselongdouble -Uuse64bitall -Uusemymalloc -Duseperlio -Dprefix=~/Sandpit/snap5.9.x-v5.13.5-520-g023257b -Uusevendorprefix -Uvendorprefix=~/Sandpit/snap5.9.x-v5.13.5-520-g023257b -Dinstallman1dir=none -Dinstallman3dir=none -Uuserelocatableinc -Umad -Accccflags-DPURIFY -de'
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=undef, usemultiplicity=undef
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=define, use64bitall=undef, uselongdouble=define
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='ccache gcc', ccflags ='-fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-Os',
    cppflags='-fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include'
    ccversion='', gccversion='4.3.2', gccosandvers=''
    intsize=4, longsize=8, ptrsize=8, doublesize=8, byteorder=12345678
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16
    ivtype='long', ivsize=8, nvtype='long double', nvsize=16, Off_t='off_t', lseeksize=8
    alignbytes=16, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags =' -fstack-protector -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib /lib64 /usr/lib64
    libs=-lnsl -ldb -ldl -lm -lcrypt -lutil -lc
    perllibs=-lnsl -ldl -lm -lcrypt -lutil -lc
    libc=/lib/libc-2.7.so, so=so, useshrplib=false, libperl=libperl.a
    gnulibc_version='2.7'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,-E'
    cccdlflags='-fPIC', lddlflags='-shared -Os -L/usr/local/lib -fstack-protector'

Locally applied patches:
    


@INC for perl 5.13.5:
    lib
    /home/nick/Sandpit/snap5.9.x-v5.13.5-520-g023257b/lib/perl5/site_perl/5.13.5/x86_64-linux-ld
    /home/nick/Sandpit/snap5.9.x-v5.13.5-520-g023257b/lib/perl5/site_perl/5.13.5
    /home/nick/Sandpit/snap5.9.x-v5.13.5-520-g023257b/lib/perl5/5.13.5/x86_64-linux-ld
    /home/nick/Sandpit/snap5.9.x-v5.13.5-520-g023257b/lib/perl5/5.13.5
    .


Environment for perl 5.13.5:
    HOME=/home/nick
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/nick/bin:/usr/local/bin:/usr/bin:/bin:/usr/games:/usr/local/sbin:/sbin:/usr/sbin
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @cpansprout

Fixed by 1be7d6f30 for threaded builds\.

In non\-threaded builds\, something else is leaking\, too\.

**Comment by p5pRT** [comment]
### From @cpansprout

On Thu Nov 25 09&#8203;:48&#8203;:33 2010\, sprout wrote&#8203;:
> Fixed by 1be7d6f30 for threaded builds\.
> 
> In non\-threaded builds\, something else is leaking\, too\.

Fixed by 78da76255\.

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'new' to 'resolved'
